app/controllers/driver_controller.rb

```diff
@@ -0,0 +1,8 @@
+class DriverController < ApplicationController
+  def index
+    query = Trip.by_user(current_user.id)
+    query = query.select { |elm| elm.date_valid? } if params[:filter] == "done"
```

Comment:
this trick is good, but hard to maintain. What if, for example, I want to search for trips in a given (past or future) date range? Maybe You can scale this to support future enhancements splitting the business logic and the param management into two separate steps. In the model you could have a scope or two to search by date and here just the logic to call it/them (I say 1 or 2 because there are two possible solutions to this, I let you find at least one of them 😈  )